openmp/libomptarget/src/PluginManager.cpp

```diff
@@ -203,6 +208,24 @@ void PluginManager::registerLib(__tgt_bin_desc *Desc) {
   for (int32_t i = 0; i < Desc->NumDeviceImages; ++i)
     PM->addDeviceImage(*Desc, Desc->DeviceImages[i]);
 
+  // Initialize any needed plugins using the image metadata if needed.
+  for (auto &R : PM->pluginAdaptors()) {
+    if (R.is_plugin_active())
+      continue;
+
+    // We can skip initializing this image if there are no images for it.
+    for (DeviceImageTy &DI : PM->deviceImages()) {
```

Comment:
Technically we'd only need to iterate over the new ones, doubtful it will ever matter.

Reply:
True, can just get them from the image passed in.

Reply:
Realized that it's a pain because it requires parsing the offloading binary format. I'm going to change that soon now that we just use the ELF.